lld/include/lld/Common/Filesystem.h

```diff
@@ -10,11 +10,15 @@
 #define LLD_FILESYSTEM_H
 
 #include "lld/Common/LLVM.h"
+#include "llvm/Support/raw_ostream.h"
+#include <memory>
 #include <system_error>
 
 namespace lld {
 void unlinkAsync(StringRef path);
 std::error_code tryCreateFile(StringRef path);
+std::unique_ptr<llvm::raw_fd_ostream> openFile(StringRef file);
```

Comment:
Could we have the refactoring of `openFile` and `openLTOOutputFile` in a separate preceding commit?